### PR TITLE
test: add unit tests to hydrate/values

### LINKED ISSUE #447: 🛠 Tooling: Add more unit testing around src/hydrate
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

#441 added a bunch of code under `src/hydrate/`. But it didn't completely unit test everything! Shame on me. 😄 

I'm filing this tracking issue for adding more unit test coverage to files in that folder. Accepting PRs that add any amount of tests - you don't have to unit test everything. In fact, I'd recommend starting with simpler functions first.

### Additional Info

Tip: for the more complex functions, see how other more-complex functions are tested.

It may not be worth unit testing `src/hydrate/index.ts` itself or the other more complex functions. 100% unit test coverage can be an annoying metric to go for.

This is a sibling issue to #471.

### PR DESCRIPTION
**Pull request by RNR1**
## PR Checklist

- [x] Addresses an existing open issue: fixes #447
- [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/template-typescript-node-package/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

An ongoing PR for adding unit tests to hydrate components.
currently added:
- `src/hydrate/values/readAuthorIfExists`
- `src/hydrate/values/readOwnerFromGitRemote`
- `src/hydrate/values/readFundingIfExists`

### CODE REVIEW

File: src/hydrate/values/readFundingIfExists.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import fs from "node:fs/promises";
+import { promises as fs } from "node:fs";
```
Comment (JoshuaKGoldberg) [inline src/hydrate/values/readFundingIfExists.ts:1]: [Question] Any particular reason for this? I've been moving towards `node:fs/promises` just to not have to use the `as`, but that's not a hard preference on my end.

**Review by JoshuaKGoldberg — APPROVED** [review]
💯 this looks great, thanks!

Just the one non-blocking curiosity question - but I'm happy to merge past it if it's a preference on your end.

File: src/hydrate/values/readFundingIfExists.ts (hunk 1 above)
Comment (RNR1) [reply src/hydrate/values/readFundingIfExists.ts:1]: @JoshuaKGoldberg The reason was that any other attempt to mock fs/promises has failed for me, until this attempt succeeded.
It seems to be a common issue when trying to mock "fs" with testing libraries

Comment (JoshuaKGoldberg) [reply src/hydrate/values/readFundingIfExists.ts]: Got it - I pushed up 9520ed8 just to show an example. Time to merge! 🚀

Comment (RNR1) [reply src/hydrate/values/readFundingIfExists.ts]: That was convenient!

**Comment by JoshuaKGoldberg** [comment]
@all-contributors please add @rnr1 for test.

> 🤖 Beep boop! This comment was added automatically by [all-contributors-auto-action](https://github.com/marketplace/actions/all-contributors-auto-action).
> Not all contributions can be detected from Git & GitHub alone. Please comment any missing contribution types this bot missed.
> ...and of course, thank you for contributing! 💙

**Pull request merged**